### ISSUE TITLE
ResizeOptions instance can't be reused

### ISSUE THREAD

**Issue by bvandier** [SixLabors/ImageSharp#473]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When reusing an instance of the ResizeOptions on two images, one landscape and one portrait, only the first image is resized correctly. This was unexpected behaviour for me, although it is working as programmed, see [https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Processing/Transforms/Resize.cs](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Processing/Transforms/Resize.cs)

### Steps to Reproduce
1. Initialize a ResizeOptions instance.
2. Use that instance to resize two different images. This behaviour is easiest to observe when the images have different orientations.

### Thoughts
I don't know if this behavior is intended. If it is, this issue is moot, though it would be nice to have this documented somewhere.

**Comment by JimBobSquarePants** [comment]
Hi @bvandier 

I've been meaning to refactor that particular section of code for a while now. Once #474 is merged the options will be reusable.

Cheers

James

**Comment by bvandier** [comment]
Thanks for taking the time to fix this!
